### PR TITLE
Jest Set Up

### PR DESCRIPTION
**Pull request by acflinch**
Jest Set Up for JS unit testing

### CODE REVIEW

File: code/frontend/node_modules/.cache/default-development/3.pack
Comment (jshilts657) [inline code/frontend/node_modules/.cache/default-development/3.pack:1]: This dir should not be included update the .gitignore and remove this dir

File: code/frontend/src/tests/dumb.test.js
Comment (jshilts657) [inline code/frontend/src/tests/dumb.test.js:1]: Are we keeping this file? Why? I forget sorry

File: .github/workflows/current-workflow.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
       - name: Check out repository code
         uses: actions/checkout@v4
       - run: echo "The repository ${{ github.repository }} has been cloned to runner."
-      - run: echo "Workflow is now ready to test code on runner."
+      - run: echo "Workflow is now ready to tests code on runner."
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: tests -> test

File: .github/workflows/current-workflow.yml [hunk 2]
```diff
@@ -60,7 +60,7 @@ jobs:
       - name: Checking out repository code
         uses: actions/checkout@v4
       - run: echo "The repository ${{ github.repository }} has been cloned to runner."
-      - run: echo "Workflow is now ready to test code on runner."
+      - run: echo "Workflow is now ready to tests code on runner."
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: tests -> test

File: .github/workflows/current-workflow.yml [hunk 3]
```diff
@@ -87,7 +87,7 @@ jobs:
...
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: tests -> test

**Review by jamzheng109 — APPROVED** [review]
looks good

**Review by jshilts657 — CHANGES_REQUESTED** [review]
You might want to try to remerge dev into this branch it seems out of date

Let me know if you need help with any requested changes.

File: .github/workflows/current-workflow.yml (hunk 1 above)
Comment (acflinch) [reply .github/workflows/current-workflow.yml]: that red mean it's been changed to the green, i believe we wanted the the test folder to be called tests

**Review by jshilts657 — APPROVED** [review]
Changes made
Looks good

**Review by rzheng2019 — APPROVED** [review]

**Pull request merged**